cmd/maestro/environments/e_development.go

```diff
@@ -16,6 +18,11 @@ func (e *devEnvImpl) VisitDatabase(c *Database) error {
 	return nil
 }
 
+func (e *devEnvImpl) VisitMessageBroker(c *MessageBroker) error {
+	c.CloudEventsSourceOptions = mqttoptions.NewSourceOptions(e.env.Config.MessageBroker.MQTTOptions, "maestro-development")
```

Comment:
most config can occur via flags, like this one. The only difference between this and the other MQTT in this PR is a param value.  This is better set by flag.

The visitor pattern in the env package is meant for complex overrides of behavior and injection of implementations.  Our [FedRAMP implementation](https://gitlab.cee.redhat.com/service/uhc-account-manager/-/blame/master/pkg/environments/e_fedramp.go#L40), for example, configures a complex service with its own dependencies and injects that. It's not just setting a flag.

Overall, we can look to set nearly all values for an environment in the `Flags()` func in this same file.  We then deploy this service with an easy `OCM_ENV` flag  (we can/should change the name of this constant).

Reply:
yes, I am suggesting that the config is better to apply here:  https://github.com/openshift-online/maestro/blob/main/cmd/maestro/environments/e_production.go#L39

we can configure each environment with defaults per that code and deploy with a simple "env var". 

the visitor pattern here is to override default _behavior_ with different implementations.  You can see that in the FedRAMP example I linked to above.

using visitor to set params on the mqtt server is incorrect in this context.

Reply:
currently no specific flags about MQTT need to be set for different environments, except for the MQTT root CA and client certificate/key. These flags pertain to file paths, but implementing them in the flags map may require additional guidance.

Reply:
done, thanks @markturansky

Reply:
this simple config should move to `pkg/config/message_broker.go`.   the visitor pattern here is intended to override _behavior_, not provide config.

Reply:
done. removed this from visit config.